src/general/density_calculators.jl

```diff
@@ -35,6 +35,16 @@ end
     return v[end, particle]
 end
 
+# *Note* that these functions are intended to internally set the density for buffer particles
+# and density correction. It cannot be used to set up an initial condition,
+# as the particle density depends on the particle positions.
+
+@inline set_particle_density(particle, v, ::SummationDensity, system, density) = system
```

Comment:
```suggestion
# WARNING!
# These functions are intended to be used internally to set the density
# of newly activated particles in a callback.
# DO NOT use outside a callback. OrdinaryDiffEq does not allow changing `v` and `u`
# outside of callbacks.
@inline set_particle_density!(v, system, ::SummationDensity, particle, density) = v
```
1. Made this a more aggressive warning.
2. This is only allowed to be used inside callbacks.
3. Should have a `!` appended.
4. Changed order of arguments to be consistent with other functions. (And it should now return `v`, as I made it the first argument.)